packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -41,6 +41,22 @@ exports.setBoundActionCreators = actions => {
   boundActionCreators = actions
 }
 
+/// Plugin options are loaded onPreInit in gatsby-node
+let pluginOptions = {
+  useMozJpeg: process.env.GATSBY_JPEG_ENCODER === `MOZJPEG`,
+  stripMetadata: true,
+}
+exports.setPluginOptions = options => {
+  // Typechecks
+  if (typeof options.useMozJpeg === `boolean`) {
```

Comment:
```js
const pluginDefaults = {
  useMozJpeg: process.env.GATSBY_JPEG_ENCODER === `MOZJPEG`,
  stripMetadata: true,
}

exports.setPluginOptions = opts => {
  const options = Object.assign({}, pluginOptions, opts);
}
```

Reply:
You might be using the wrong language if you want strict type-checking :)

I really wouldn't worry too much about it. I do think you raise a good point--I've wanted to implement an API using something like `yup` or `joi` for quite a while, but just haven't gotten around to doing so yet. That being said, if we think of the "false positive" case here (e.g. a string is passed, or undefined, etc.) I think we're pretty isolated from any errors since we're just checking for truthy/falsy rather than using these values in a meaningful way--e.g. passing them to something that needs more type safety.